src/qibo/gates/abstract.py

```diff
@@ -1,11 +1,15 @@
 import collections
+import json
 from typing import List, Sequence, Tuple
 
 import sympy
 
 from qibo.backends import GlobalBackend
 from qibo.config import raise_error
 
+REQUIRED_FIELDS = ["name", "init_kwargs", "_target_qubits", "_control_qubits"]
+REQUIRED_FIELDS_INIT_KWARGS = ["theta", "phi", "lam"]
```

Comment:
You could include the `trainable` here, right? Might be relevant for some applications, but I am not sure if those would use the serialization anyway.

Reply:
Yeah, I don't know either, for what I needed it wasn't necessary. This can extended as the application that requires it appears. I would try to save the least for now.

Reply:
Agreed, let's stick to something simple.

I want to rework the gates anyhow, and (if possible) write a simpler and more robust serialization. Maybe we will need to improve this in any case, but maybe not.